### PR TITLE
Remove tooling/cpu_features, switch to base/internal/cpu.

### PR DESCRIPTION
**Pull request by bjacob**
Fix a bug in the ARM64 path in base/internal/cpu, flagged by the
warnings pasted below. The issue was some missing parentheses, but I
decided to reword this as an if() statement to make it less tricky.
```
/usr/local/google/home/benoitjacob/iree/runtime/src/iree/base/internal/cpu.c:43:3: error: operator '?:' has lower precedence than '|'; '|' will be evaluated first [-Werror,-Wbitwise-conditional-parentheses]
  IREE_SET_IF_HWCAP(hwcap, IREE_HWCAP_ASIMDDP, out_fields[0],
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/google/home/benoitjacob/iree/runtime/src/iree/base/internal/cpu.c:28:23: note: expanded from macro 'IREE_SET_IF_HWCAP'
                      ? field_bit
```

Change the e2e matmul tests YAML trace format to not have "+" prefixes
on the required target features. That was done out of lazyness as we had
the "+" in the LLVM cpu features argument, so the old
tooling/cpu_features accepted +-prefixed feature keys, but the new
base/internal/cpu does not.

### CODE REVIEW

**Review by benvanik — APPROVED** [review]
nice! thanks!!

File: tools/iree-e2e-matmul-test.c [hunk 1]
```diff
@@ -1056,17 +1056,18 @@ static iree_status_t replay_event_call_matmul(iree_trace_replay_t* replay,
...
        item != target_features_node->data.sequence.items.top; ++item) {
     yaml_node_t* item_node = yaml_document_get_node(document, *item);
     iree_string_view_t required_feature = iree_yaml_node_as_string(item_node);
     if (iree_string_view_is_empty(required_feature)) continue;
-    bool feature_is_supported = false;
-    IREE_RETURN_IF_ERROR(iree_cpu_features_query(cpu_features, required_feature,
-                                                 &feature_is_supported));
+    int64_t feature_is_supported = 0;
+    fprintf(stderr, "Looking up feature by key: %.*s\n",
+            (int)required_feature.size, required_feature.data);
```
Comment (ScottTodd) [inline tools/iree-e2e-matmul-test.c:1068]: Probably want to remove this `fprintf`, so it doesn't appear in test / CI output for every run?

Comment (bjacob) [reply tools/iree-e2e-matmul-test.c:1068]: woops, thanks for the catch!

Comment (bjacob) [reply tools/iree-e2e-matmul-test.c]: done

**Review by ScottTodd — APPROVED** [review]
